doc/source/rllib/rllib-replay-buffers.rst

```diff
@@ -0,0 +1,145 @@
+For your algorithms this means that buffers with a high capacity give the opportunity to learn from older samples, while smaller buffers
+make the learning process more on-policy. An exception from this strategy is made in buffers that implement `reservoir sampling <https://www.cs.umd.edu/~samir/498/vitter.pdf>`__.
+
+
+Replay Buffers in RLlib
+=======================
+
+RLlib comes with a set of extendable replay buffers built in.
+We provide a base :py:class:`~ray.rllib.utils.replay_buffers.replay_buffer.ReplayBuffer` class from which you can build your own buffer.
+In most algorithms, we require :py:class:`~ray.rllib.utils.replay_buffers.multi_agent_replay_buffer.MultiAgentReplayBuffer`\s.
+This is because we want them to generalize to the the multi-agent case. Therefore, these buffer's ``add()`` and ``sample()`` methods require a ``policy_id`` to handle experiences per policy.
+Have a look at the :py:class:`~ray.rllib.utils.replay_buffers.multi_agent_replay_buffer.MultiAgentReplayBuffer` to get a sense of how it extends our base class.
+You can find buffer types and arguments to modify their behaviour as part of RLlib's default parameters. They are part of
+the ``replay_buffer_config``.
+
+Basic Usage
+-----------
+
+You will rarely have to define your own replay buffer sub-class, when running an experiment, but rather configure existing buffers.
+The following is `from RLlib's examples section <https://github.com/ray-project/ray/blob/master/rllib/examples/replay_buffer_api.py>`__:  and runs the R2D2 algorithm with `PER <https://arxiv.org/abs/1511.05952>`__ (which by default it doesn't).
+The highlighted lines focus on the PER configuration.
+
+.. dropdown:: **Executable example script**
+    :animate: fade-in-slide-down
+
+    .. literalinclude:: ../../../rllib/examples/replay_buffer_api.py
+        :emphasize-lines: 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77
+        :language: python
+        :start-after: __sphinx_doc_replay_buffer_api_example_script_begin__
+        :end-before: __sphinx_doc_replay_buffer_api_example_script_end__
+
+.. tip:: Because of its prevalence, most Q-learning algorithms support PER. The priority update step that is needed is embedded into their training iteration functions.
+.. warning:: If your custom buffer requires extra interaction, you will have to change the training iteration function, too!
+
+
+Specifying a buffer type works the same way as specifying an exploration type.
+Here are three ways of specifying a type:
+
+.. dropdown:: **Changing a replay buffer configuration**
+    :animate: fade-in-slide-down
+
+    .. literalinclude:: ../../../rllib/examples/documentation/replay_buffer_demo.py
+        :language: python
+        :start-after: __sphinx_doc_replay_buffer_type_specification__begin__
+        :end-before: __sphinx_doc_replay_buffer_type_specification__end__
+
+Apart from specifying a type, other parameters in that config are:
```

Comment:
Maybe mention here that all the other args (non-type) are basically c'tor kwargs of the used type.

I also think we should mention `capacity` first, as it's the most important one. Then maybe storage unit, and only then the more exotic ones like worker_side_prio, etc..

Reply:
But that would not be correct! There are non-ctor args in there as well. Like worker_side_prioritization.
Capacity is currently the first example of a c'tor arg.

I've changed the this part a bit - lmk if this still needs improvement :)